File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for p in s:
        if p == '(':
            stack.append(p)
        else:
            if stack and stack[-1] == '(':
                stack.pop()
            else:
                stack.append(p)
                
    if not stack: return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
    const stack = [];
    for (p of s) {
        if (p === '(') {
            stack.push('(');
        } else if (stack.length > 0 && stack[stack.length - 1] === '(') {
            stack.pop();
        } else {
            return false;
        }
    }
    
    if (stack.length === 0)
        return true;
    else
        return false;
}